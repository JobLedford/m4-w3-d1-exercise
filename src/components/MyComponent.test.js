import React from 'react';
import { shallow } from 'enzyme';
import MyComponent from './MyComponent';

describe("MyComponent", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<MyComponent />);
  });
  it("should render intial layout", () => {
    //when
    const componentTA = shallow(<MyComponent />);
    //then
    expect(componentTA.getElements()).toMatchSnapshot()
  });
  it("should create an entry in component state", () => {
    //given
    const componentTB = shallow(<MyComponent/>)
    const form = componentTB.find('input');
    form.props().onChange({target:{
      name: 'myName',
      value: 'myValue'
    }});
    //then
    //expect(componentTB.state('input')).toBeDefined();
    expect(componentTB.state('input')).toEqual('myValue');
  });
});
