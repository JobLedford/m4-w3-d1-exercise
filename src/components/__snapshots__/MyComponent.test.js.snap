// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render intial layout 1`] = `
Array [
  <div>
    <input
      onChange={[Function]}
      type="text"
    />
  </div>,
]
`;
